crates/polars-ops/src/chunked_array/strings/split.rs

```diff
@@ -30,10 +33,16 @@ where
                 },
                 Some(s) => {
                     let mut arr_iter = arrs.iter_mut();
-                    let split_iter = op(s, by);
-                    (split_iter)
-                        .zip(&mut arr_iter)
-                        .for_each(|(splitted, arr)| arr.push(Some(splitted)));
+                    if is_empty_pattern && split_exact {
```

Comment:
Don't do this as part of the inner loop, make the decision once.

Reply:
I moved the check out of the inner loop in 273b37a306293af191340b03ce1d1ddb2ebb2e58.